src/main/webapp/js/http.js

```diff
@@ -1,15 +1,22 @@
-function httpRequestSync(method, url, body) {
-    var xhr = new XMLHttpRequest();
-    xhr.open(method, url, false); // false for synchronous request
-    xhr.setRequestHeader('Content-Type', 'application/json');
-    if (method === 'POST' && body) {
-        xhr.send(body);
-    } else {
-        xhr.send();
-    }
-    if (xhr.status === 200) {
-        return xhr.responseText;
-    } else {
-        throw new Error('Request failed with status ' + xhr.status);
+async function asyncHttpRequest(method = 'GET', url, body = null, callback) {
```

Comment:
This has a different amount of arguments compared to its native java counterpart?

Reply:
For some reason, we need two native methods and CreateAsyncHttpRequest is the native method that calls js script and then we use the other method in the java code

Reply:
My point it that this:
public static native String asyncHttpRequest(String method, String url, JSObject body);
has fewer arguments than this:
async function asyncHttpRequest(method = 'GET', url, body = null, callback)

Reply:
You are correct but we don't need to provide the callback function as input parameter when calling asyncHttpRequest method.